src/collision.cpp

```diff
@@ -419,8 +420,9 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 	// A fairly large value in here makes moving smoother
 	//f32 d = 0.15*BS;
 
-	// This should always apply, otherwise there are glitches
-	assert(d > pos_max_d);	// invariant
+	f32 d = 0.0f;	// Temporary fix, any nonzero d causes collision glitches, the more the greater it is.
+	// ultimately it has to be determined if any uncertainty is involved, and if it is, eliminated
+	// and d & pos_max_d params removed from function calls.
```

Comment:
None of this applies anymore does it?

Reply:
I intended too keep it until the tests gone ok, as a reminder to update the collisionMoveSimple parameter list everywhere it's called, is that ok?

Reply:
Just make sure to remove it before merge.

Reply:
removed the comment, the d stays because it's used by wouldCollideWithCeiling() and in touching_ground determination.